modules/firemission/Dia/Dia_Global.sqf

```diff
@@ -0,0 +1,243 @@
+	private _roundsRequired = _this select 2;
+	_unit setVariable [VAR_SART_ARTROUNDSFIRED,[_roundsFired,_roundsRequired],true];
+
+};
+
+FNC_GetArtyCaller =
+{
+	private _unit = _this;
+	private _callerUnit = _unit getVariable [VAR_SART_CALLER,objNull];
+	_callerUnit
+};
+FNC_GetArtyCallerText =
+{
+	private _unit = _this;
+	private _caller = _unit call FNC_GetArtyCaller;
+	private _returnName = "Non specified";
+	if(!(isNull  _caller)) then
+	{
+		_returnName = name _caller;
+	};
+
+	_returnName
+};
+FNC_SetArtyCaller =
+{
+	private _unit = _this select 0;
+	private _caller = _this select 1;
+	_unit setVariable [VAR_SART_CALLER,_caller,true];
+};
+
+FNC_GetArtyDisplayName =
+{
+	private _unit = _this;
+	private	_text = _unit getVariable [VAR_SART_ARTCUSTOMNAME,""];
+	if(_text == "") then { _text = getText (configfile / "CfgVehicles" /  typeOf(_unit) / "displayName") };
+	_text
+};
+FNC_GetAmmoDisplayNameAndIndex =
+{
+	private _unit =_this;
+	private _ret = [];
+	private _possibleMags = getArray (configfile >> "CfgWeapons">>  (((_unit) weaponsTurret [0]) select 0) >> "magazines");
+	{
+		_ret pushBack [_forEachIndex,_x,getText (configfile >> "CfgMagazines">>  _x >> "displayName")];
+	}forEach _possibleMags;
+	_ret
+};
+FNC_GetArtyIndexAmmoClassname =
+{
+		private _unit =_this select 0;
+		private _index =_this select 1;
+		private _ret = ((_unit call FNC_GetArtyAmmo) select _index) select 0;
+		_ret
+};
+FNC_GetArtyAmmo =
+{
+	private _unit =_this;
+	private _ammo = magazinesAmmo _unit;
+	private _ret = [];
+	private _possibleMags = getArray (configfile >> "CfgWeapons">>  (((_unit) weaponsTurret [0]) select 0) >> "magazines");
+	{
+		_ret pushBack [_x,0];
+	}forEach _possibleMags;
+
+	{
+		_check = _x;
+		_found = false;
+		{
+			if(_x select 0 == _check select 0) then
+			{
+				_found = true;
+				_num = (_check select 1) + (_x select 1);
+				_x set [1,_num];
+			}
+		}forEach _ret;
+		if(!_found) then
+		{
+			hint "GetArtyAmmo Error";
+		};
+
+	}forEach _ammo;
+	_ret
+};
+
+FNC_ArtLoadAviableArtilleries =
+{
+			private _id = _this select 0;
+			private _shellSelect = _this select 1;
+			private _guns = player getVariable [VAR_SART_OBSGUNS,[]];
+			private _usableGuns = [];
+			{
+				if(_x call FNC_IsArtyAviable) then
+				{
+					_usableGuns pushBack _x;
+				};
+			}forEach _guns;
+
+
+			lbClear _id;
+			{
+				lbAdd [_id,_x call FNC_GetArtyDisplayName];
+
+			}forEach _usableGuns;
+			lbSetCurSel [_id,0];
+};
+FNC_ArtSetArtillery =
+{
+	private _id = _this select 0;
+	private _gun = _this select 1;
+	if(_gun >= 0) then
+	{
+		_guns = player getVariable [VAR_SART_OBSGUNS,[]];
+		_usableGuns = [];
+		{
+			if(_x call FNC_IsArtyAviable) then
+			{
+				_usableGuns pushBack _x;
+			};
+		}forEach _guns;
+		_actualGunUnit = _usableGuns select _gun;
+
+		_ammo = _actualGunUnit call FNC_GetArtyAmmo;
+
+		{
+			_text = (str (_x select 1)) + "x " + getText (configfile / "CfgMagazines" / (_x select 0) / "displayName");
+			lbAdd [_id, _text];
```

Comment:
You never clear this list. Repeatedly switching between guns will make it grow really long.